lib/grpc-connector.js

```diff
@@ -128,7 +129,15 @@ GRPCConnector.prototype.setupDataAccessObject = function() {
 
 function createMethod(client, name, method, remoting) {
   var fn = function() {
-    return client[name].apply(client, arguments);
+    const args =
+      arguments.length === 1 ? [arguments[0]] : Array.apply(null, arguments);
+    const lastArg = args.length ? args[args.length - 1] : undefined;
+    const callback = typeof lastArg === 'function' ? lastArg : undefined;
+    if (callback) {
+      client[name].apply(client, arguments);
+    } else {
+      return promisify(client[name]).apply(client, arguments);
```

Comment:
Calling `promisify` on each function call is suboptimal performance-wise. 

So far, we were using a different approach based on creating a callback wrapping a new promise, see  juggler's [createPromiseCallback](https://github.com/strongloop/loopback-datasource-juggler/blob/1ef38a6929ae100c73030b720097ba5c4b90a7b3/lib/utils.js#L550-L560). Feel free to copy the full source code of that function into this project.

```js
if (callback) {
  client[name].apply(client, arguments);
} else {
  callback = createPromiseCallback();
  args.push(callback);
  client[name].apply(client, args);
  return callback.promise;
}
```

Alternatively, if you prefer to leverage `util.promisify`, then please cache the promisified function. For example:

```js
if (callback) { 
  client[name].apply(client, arguments);
} else {
  if (!client[`_${name}Async`]) {
    client[`_${name}Async`] = promisify(client[name]);
  }
  client[`_${name}Async`].apply(client, arguments);
}
```